loader/src/main/java/net/neoforged/neoforgespi/locating/IModFileReader.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforgespi.locating;
+
+import cpw.mods.jarhandling.JarContents;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Inspects {@link JarContents} found by {@link IModFileCandidateLocator} and tries to turn them into {@link IModFile}.
+ * <p>
+ * Picked up via ServiceLoader.
+ */
+public interface IModFileReader {
+    /**
+     * Provides a mod from the given {@code jar}. Any thrown exception will be reported in relationship to the given
+     * jar contents.
+     *
+     * @param jar        the mod jar contents
+     * @param attributes The attributes relating to this mod files discovery.
+     * @return {@code null} if this provider can't handle the given jar, otherwise a result indicating success or failure.
```

Comment:
we don't have results here anymore?

Reply:
I was commenting on the `otherwise a result indicating success or failure` javadoc, that could have been clearer. I agree that an exception is cleaner.